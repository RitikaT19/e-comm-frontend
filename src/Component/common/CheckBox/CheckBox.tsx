import React from "react";
// import "../../styles/checkbox.css";
interface Props {
  label?: string;
  checked?: boolean;
  id: string;
  handleChange:
    | ((event: React.ChangeEvent<HTMLInputElement>) => void)
    | undefined;
  value?: string;
}

export const CheckBox: React.FC<Props> = ({
  label,
  id,
  checked,
  handleChange,
  value,
}) => {
  return (
    <label className="container-checkbox" htmlFor={id + "_element"}>
      <input
        type="checkbox"
        name="check"
        onChange={handleChange}
        id={id + "_element"}
        checked={checked}
        value={value}
      />
      <span className="label-text">{label}</span>
    </label>
  );
};
