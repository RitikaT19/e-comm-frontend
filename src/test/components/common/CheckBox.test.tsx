import React, { Component } from "react";
import { mount, ReactWrapper } from "enzyme"; //import mount and ReactWrapper for testing
//import component that is to be tested
import { CheckBox } from "../../../component/common/CheckBox/CheckBox";

describe("Testing CheckBox Component", () => {
  //create a wrapper object
  let wrapper: ReactWrapper<any, Readonly<{}>, Component<{}, {}, any>>;
  //the default props that this component expects
  const defaultProps = {
    id: "test",
    handleChange: jest.fn(),
    label: "test text field",
    checked: true,
  };
  //mount the component and store it in the wrapper before each test
  beforeEach(() => {
    wrapper = mount(<CheckBox {...defaultProps} />);
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should should have one input field", () => {
    expect(wrapper.find("input")).toHaveLength(1);
  });

  it("should should have one label", () => {
    expect(wrapper.find("label")).toHaveLength(1);
  });

  it("should have the id same as the id+_element prop", () => {
    expect(wrapper.find("input").props().id).toBe(defaultProps.id + "_element");
  });

  it("should call the handle change prop when a user types in the input field", () => {
    const event = { target: { value: "test input" } };
    wrapper.find("input").simulate("change", event);
    expect(defaultProps.handleChange).toHaveBeenCalledTimes(1);
  });
});
