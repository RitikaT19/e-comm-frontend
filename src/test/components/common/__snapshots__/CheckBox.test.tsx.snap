// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CheckBox Component should match snapshot 1`] = `undefined`;
